src/app/reducers/incidents/incidentSlice.js

```diff
@@ -62,6 +62,55 @@ export const errorLog = state => state.incidents.meta.errorLog;
 export const refreshState = state => state.incidents.meta.refresh;
 // For getting the value of the current state throughout the app;
 
+
+// Formatter Function for preparing selectedIncident Data for DetailsBlock
+// This moves the responsibility out of the component, and into the reducer.
+// Transformation and formatting of data makes more sense in reducer middleware than the React component. (eg: Smart vs Dumb components)
+export const getDetailBlocks = (state) => {
```

Comment:
This function acts as reducer middleware to take data in one structure (a raw API response, for example) and format it to a different structure (a clean array of objects for a UI table or list). In this case, we're formatting bits of data from our Store and combining it in a way that it easier for **DetailsBlock** to consume and display various Incident Data without making **DetailsBlock** worry about formatting timestamps and whatnot.

This makes **DetailsBlocks**' job as simple as: "Just display the data I give you, I've already formatted it for you."

This way, our presentational components are more concerned about Markup and Styles, and less about combining arrays and changing data values. That's what reducers are for.

If Redux is the "Brains" of an application, then React is the "Body".